### PR TITLE
unify composition type errors

### PR DESCRIPTION
**Pull request by diogob**
This should improve readability for complex types that have to validate composition.

It also eliminates the redundancy of two types used to represent composition failure, using only a new version of `FailToCompose`.

Reuse the `IncompatibleArguments` just to test for errors.

### CODE REVIEW

File: src/combinators.ts [hunk 1]
```diff
@@ -92,7 +92,7 @@ function all<Fns extends Composable[]>(...fns: Fns) {
  */
 function collect<Fns extends Record<string, Composable>>(fns: Fns) {
   const fnsWithKey = Object.entries(fns).map(([key, cf]) =>
-    map(cf, (result) => ({ [key]: result })),
+    map(cf, (result) => ({ [key]: result }))
```
Comment (diogob) [inline src/combinators.ts:95]: this file has just left-over formatting changes

File: src/environment/tests/pipe.test.ts [hunk 2]
```diff
@@ -158,7 +159,7 @@ describe('pipe', () => {
     type _FN = Expect<
       Equal<
         typeof fn,
-        ['Fail to compose', undefined, 'does not fit in', boolean]
+        Internal.FailToCompose<undefined, boolean>
```
Comment (diogob) [inline src/environment/tests/pipe.test.ts:162]: this will make it easier to change the shape of composition failure

File: src/internal/types.ts [hunk 3]
```diff
@@ -1,18 +1,19 @@
 // deno-lint-ignore-file no-namespace
 
 namespace Internal {
-  export type IncompatibleArguments<A, B> = {
+  export type IncompatibleArguments = {
```
Comment (diogob) [inline src/internal/types.ts:4]: no need to have two separate composition failure types, I have repurposed this one to use in type ternary tests

**Review by gustavoguichard — APPROVED** [review]
Awesome! 👏🏽

**Pull request merged**